pkg/spec3/serialization.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spec3
+
+import (
+	"github.com/go-openapi/jsonreference"
+	"k8s.io/kube-openapi/pkg/validation/spec"
+)
+
+func refFromMap(r *spec.Ref, v map[string]interface{}) error {
```

Comment:
I duplicated these functions for now. Still thinking about the best way to handle these

Reply:
could some of them go into internal as Jordan suggested?

Reply:
There's an import cycle with the internal package depending on the types defined in `spec/validation` and the types in the package depending on the function defined in the internal package. We've usually handled these cycles via interfaces but that probably wouldn't work in our scenario.

Reply:
Hmm I did not think of that. Maybe refFromMap could return jsonReference. Also `sanitizeExtensions` nor `postProcessExtensions` need to operate on `spec.Extensions`: I think that type is aliased to a `map[string]any`

Reply:
Those are great suggestions, I'll give it a shot thanks!

Reply:
https://github.com/kubernetes/kube-openapi/pull/363 is updated based on the suggestions here